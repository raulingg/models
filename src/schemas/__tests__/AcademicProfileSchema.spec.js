const mongoose = require('mongoose/browser');
const AcademicProfileSchema = require('../AcademicProfileSchema')(mongoose);
const AcademicProfileTagSchema = require('../AcademicProfileTagSchema')(mongoose);
const AcademicProfileCommentSchema = require('../AcademicProfileCommentSchema')(mongoose);
const UserSchema = require('../UserSchema')(mongoose);
const TagSchema = require('../TagSchema')(mongoose);

describe('AcademicProfileSchema', () => {
  it('should fail with empty embedded documents', (done) => {
    const tag = new mongoose.Document({}, AcademicProfileTagSchema);
    const comment = new mongoose.Document({}, AcademicProfileCommentSchema);

    const doc = new mongoose.Document({
      tags: [tag],
      comments: [comment],
    }, AcademicProfileSchema);

    return doc.validate((err) => {
      expect(err).toMatchSnapshot();
      done();
    });
  });

  it('should successfully validate with proper values with tag', (done) => {
    const createdBy = new mongoose.Document({
      uid: 'xxxxxxxxxx',
      email: 'foo@bar.baz',
    }, UserSchema);

    const tag = new mongoose.Document({}, TagSchema);
    const academicProfileTag = new mongoose.Document({
      createdBy: createdBy._id,
      tag: tag._id,
      assignmentReason: 'foo',
    }, AcademicProfileTagSchema);

    const doc = new mongoose.Document({
      tags: [academicProfileTag],
    }, AcademicProfileSchema);

    return doc.validate((err) => {
      expect(err).toBe(null);
      done();
    });
  });
});
