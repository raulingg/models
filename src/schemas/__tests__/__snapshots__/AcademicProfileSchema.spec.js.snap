// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AcademicProfileSchema should fail with empty embedded documents 1`] = `[ValidationError: Validation failed: tags.0.assignmentReason: Path \`assignmentReason\` is required., tags.0.tag: Path \`tag\` is required., tags.0.createdBy: Path \`createdBy\` is required., comments.0.text: Path \`text\` is required., comments.0.type: Path \`type\` is required., comments.0.cohortProject: Path \`cohortProject\` is required., comments.0.createdBy: Path \`createdBy\` is required.]`;
